### PR TITLE
feat(#1837): add git.disable_for_dirs

### LINKED ISSUE #1837: Disable git integration for specific paths (i.e. `$HOME`)?
### Discussed in https://github.com/nvim-tree/nvim-tree.lua/discussions/1834

<div type='discussions-op-text'>

<sup>Originally posted by **NiklasKappel** December 18, 2022</sup>
My `$HOME` is a git repository that manages my dotfiles. Necessarily, that repository contains a huge number of mostly untracked files. This causes performance issues with nvim-tree. If I run nvim from my home directory, there is a significant delay when I run nvim-tree itself or open any file after that. The delay goes away if I disable the nvim-tree git integration entirely or if I start nvim in a different, smaller git repo.

Is there a way to disable the nvim-tree git integration for a specific path such as `/home/asdf/`, but keep the feature enabled for smaller, regular repos?</div>

### PR DESCRIPTION
**Pull request by alex-courtis**
fixes #1837 

@NiklasKappel I'd be most grateful if you tested this feature:
```sh
cd /path/to/nvim-tree.lua
git pull
git checkout 1837-git-disable-dirs
```

### CODE REVIEW

File: doc/nvim-tree-lua.txt [hunk 1]
```diff
@@ -434,6 +434,7 @@ applying configuration.
         ignore = true,
         show_on_dirs = true,
         show_on_open_dirs = true,
+        disabled_dirs = {},
```
Comment (gegoune) [inline doc/nvim-tree-lua.txt:437]: Wouldn't it be more consistent with other names to call it something like `disable_for_dirs`?

Comment (alex-courtis) [reply doc/nvim-tree-lua.txt:437]: Nice. I'll update.

File: doc/nvim-tree-lua.txt [hunk 2]
```diff
@@ -710,6 +710,10 @@ Git integration with icons and colors.
     Only relevant when `git.show_on_dirs` is `true`.
       Type: `boolean`, Default: `true`
 
+    *nvim-tree.git.disabled_dirs*
+    Disable git integration when git top-level matches absolute path.
```
Comment (alex-courtis) [inline doc/nvim-tree-lua.txt:714]: This might actually work for relative paths.

TODO: test and update doc.

**Comment by alex-courtis** [comment]
> Interestingly, I can not reproduce the delay I originally described in #1837 right now, neither with the master branch nor this branch. Was there a change to how nvim-tree treats untracked files in the meantime?

Background git operation experiment was recently concluded and always enabled #2104
That may be the reason for the increase in speed.

Thank you for testing this out.

**Pull request merged**